src/Modal.js

```diff
@@ -428,11 +434,14 @@ class Modal extends React.Component {
     }
   }
 
-  handleDocumentKeyUp = (e) => {
+  handleDocumentKeyDown = (e) => {
     if (this.props.keyboard && e.keyCode === 27 && this.isTopModal()) {
-      if (this.props.onEscapeKeyUp) {
-        this.props.onEscapeKeyUp(e);
+      let escapeFunc = this.props.onEscapeKeyDown || this.props.onEscapeKeyUp;
```

Comment:
this is a bit misleading. I'd make sure the old handler actually gets called on keyUp